### PR TITLE
i18n: accept array of locales in lookupLocale

### PR DESCRIPTION
**Pull request by connorjclark**
ref https://chromium-review.googlesource.com/c/devtools/devtools-frontend/+/2382176/comment/979304b7_ee2df92f/

### CODE REVIEW

File: lighthouse-core/lib/i18n/i18n.js [hunk 1]
```diff
@@ -157,18 +157,17 @@ const formats = {
...
  * - progressively shorter prefixes (`de-CH-1996` -> `de-CH` -> `de`)
- * - the default locale ('en') if no match is found
  *
- * If `locale` isn't provided, the default is used.
+ * If `locale` isn't provided or one could not be found, undefined is returned.
  * @param {string=} locale
- * @return {LH.Locale}
+ * @return {LH.Locale|undefined}
  */
 function lookupLocale(locale) {
```
Comment (connorjclark) [inline lighthouse-core/lib/i18n/i18n.js:165]: paul suggested perhaps we should accept an array of locales here and do the "try the next value" in LH core.

File: lighthouse-core/config/config.js [hunk 2]
```diff
@@ -473,7 +473,7 @@ class Config {
     // If a locale is requested in flags or settings, use it. A typical CLI run will not have one,
     // however `lookupLocale` will always determine which of our supported locales to use (falling
     // back if necessary).
-    const locale = i18n.lookupLocale((flags && flags.locale) || settingsJson.locale);
+    const locale = i18n.lookupLocale((flags && flags.locale) || settingsJson.locale) || 'en';
```
Comment (paulirish) [inline lighthouse-core/config/config.js:476]: let's put this 'en' on `i18n.defaultLocale` instead.

File: lighthouse-core/test/lib/i18n/i18n-test.js [hunk 3]
```diff
@@ -118,16 +118,28 @@ describe('i18n', () => {
...
+
+    it('falls back to default if locale not provided or cant be found', () => {
+      expect(i18n.lookupLocale(undefined)).toEqual('en');
+      expect(i18n.lookupLocale(invalidLocale)).toEqual('en');
+      expect(i18n.lookupLocale([invalidLocale, invalidLocale])).toEqual('en');
+    });
+
     it('falls back to root tag prefix if specific locale not available', () => {
-      expect(i18n.lookupLocale('en-JKJK')).toEqual('en');
+      expect(i18n.lookupLocale('es-JKJK')).toEqual('es');
```
Comment (connorjclark) [inline lighthouse-core/test/lib/i18n/i18n-test.js:138]: test case was conflated with the default fallback behavior.

**Comment by connorjclark** [comment]
> Fun fact: both `Intl.getCanonicalLocales` and `lookupClosestLocale` already take `string|Array<string>|undefined` :)
> 
> Can we use that to reduce work and custom resolution logic here?

oooooh good catch. I tried to use this but it failed a test:

image

**Comment by brendankenny** [comment]
Wait, is there a bug in  `Intl.getCanonicalLocales`? I don't think we polyfill that...

**Comment by brendankenny** [comment]
> perhaps if lookupClosestLocale accepted string[]

I think that's just our local d.ts declaration

**Comment by brendankenny** [comment]
@connorjclark do you mind updating the docs to mention the array behavior? To clear up how prefix shortening and array order interact in precedence, for instance.

**Pull request merged**